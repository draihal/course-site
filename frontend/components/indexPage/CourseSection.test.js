import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestCoursesSection from './CoursesSection'


describe('CoursesSection', () => {
  it('render correctly CoursesSection component', () => {
    const tree = renderer.create(<TestCoursesSection
      description={'Тестовое описание'}
      number_of_students={111}
      number_of_groups={11}
      number_of_teachers={11}
    />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render CoursesSection correctly with given props', () => {
    const { queryByText } = render(<TestCoursesSection
      description={'Тестовое описание'}
      number_of_students={111}
      number_of_groups={11}
      number_of_teachers={11}
    />);
    expect(queryByText('Тестовое описание')).toBeTruthy();
    expect(queryByText('111 студентов')).toBeTruthy();
    expect(queryByText('11 групп')).toBeTruthy();
    expect(queryByText('11 преподавателей')).toBeTruthy();
  });

});