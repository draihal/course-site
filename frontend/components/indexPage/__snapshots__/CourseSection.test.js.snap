// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesSection render correctly CoursesSection component 1`] = `
<div
  className="offset"
  id="courses"
>
  <div
    className="container"
  >
    <div
      className="col-12 narrow text-center"
    >
      <h1>
        Курсы
      </h1>
      <p
        className="lead"
      >
        Тестовое описание
      </p>
      <a
        className="btn btn-secondary btn-md"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Выбрать
      </a>
    </div>
  </div>
  <div
    id="features"
  >
    <div
      className="jumbotron"
    >
      <div
        className="container"
      >
        <div
          className="narrow text-center"
        >
          <div
            className="col-12"
          >
            <h3
              className="heading"
            >
              Наша статистика
            </h3>
            <div
              className="heading-underline"
            >
               
            </div>
          </div>
          <div
            className="row text-center"
          >
            <div
              className="col-md-4"
            >
              <div
                className="feature"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user-graduate fa-w-14 fa-4x "
                  data-icon="user-graduate"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "transformOrigin": "0.4375em 0.1875em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    style={Object {}}
                    transform="translate(224 256)"
                  >
                    <g
                      style={Object {}}
                      transform="translate(0, -160)  scale(0.8125, 0.8125)  rotate(0 0 0)"
                    >
                      <path
                        d="M319.4 320.6L224 416l-95.4-95.4C57.1 323.7 0 382.2 0 454.4v9.6c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-9.6c0-72.2-57.1-130.7-128.6-133.8zM13.6 79.8l6.4 1.5v58.4c-7 4.2-12 11.5-12 20.3 0 8.4 4.6 15.4 11.1 19.7L3.5 242c-1.7 6.9 2.1 14 7.6 14h41.8c5.5 0 9.3-7.1 7.6-14l-15.6-62.3C51.4 175.4 56 168.4 56 160c0-8.8-5-16.1-12-20.3V87.1l66 15.9c-8.6 17.2-14 36.4-14 57 0 70.7 57.3 128 128 128s128-57.3 128-128c0-20.6-5.3-39.8-14-57l96.3-23.2c18.2-4.4 18.2-27.1 0-31.5l-190.4-46c-13-3.1-26.7-3.1-39.7 0L13.6 48.2c-18.1 4.4-18.1 27.2 0 31.6z"
                        fill="currentColor"
                        style={Object {}}
                        transform="translate(-224 -256)"
                      />
                    </g>
                  </g>
                </svg>
                <h3>
                  111
                   студентов
                </h3>
                <p>
                  Некоторый текст!
                </p>
              </div>
            </div>
            <div
              className="col-md-4"
            >
              <div
                className="feature"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user fa-w-14 fa-4x "
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "transformOrigin": "0.4375em 0.1875em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    style={Object {}}
                    transform="translate(224 256)"
                  >
                    <g
                      style={Object {}}
                      transform="translate(0, -160)  scale(0.8125, 0.8125)  rotate(0 0 0)"
                    >
                      <path
                        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                        fill="currentColor"
                        style={Object {}}
                        transform="translate(-224 -256)"
                      />
                    </g>
                  </g>
                </svg>
                <h3>
                  11
                   групп
                </h3>
                <p>
                  Еще какой-то текст!
                </p>
              </div>
            </div>
            <div
              className="col-md-4"
            >
              <div
                className="feature"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user-friends fa-w-20 fa-4x "
                  data-icon="user-friends"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "transformOrigin": "0.625em 0.1875em",
                    }
                  }
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    style={Object {}}
                    transform="translate(320 256)"
                  >
                    <g
                      style={Object {}}
                      transform="translate(0, -160)  scale(0.8125, 0.8125)  rotate(0 0 0)"
                    >
                      <path
                        d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
                        fill="currentColor"
                        style={Object {}}
                        transform="translate(-320 -256)"
                      />
                    </g>
                  </g>
                </svg>
                <h3>
                  11
                   преподавателей
                </h3>
                <p>
                  И еще текст!
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
